evennia/utils/validfuncs.py

```diff
@@ -0,0 +1,175 @@
+
+    for interval in time_string:
+        if _re.match(r'^[\d]+s$', interval.lower()):
+            seconds =+ int(interval.lower().rstrip("s"))
+        elif _re.match(r'^[\d]+m$', interval):
+            minutes =+ int(interval.lower().rstrip("m"))
+        elif _re.match(r'^[\d]+h$', interval):
+            hours =+ int(interval.lower().rstrip("h"))
+        elif _re.match(r'^[\d]+d$', interval):
+            days =+ int(interval.lower().rstrip("d"))
+        elif _re.match(r'^[\d]+w$', interval):
+            weeks =+ int(interval.lower().rstrip("w"))
+        elif _re.match(r'^[\d]+y$', interval):
+            days =+ int(interval.lower().rstrip("y")) * 365
+        else:
+            raise ValueError(f"Could not convert section '{interval}' to a {thing_name}.")
+
+    return _dt.timedelta(days, seconds, 0, 0, minutes, hours, weeks)
+
+
+def future(entry, thing_name="Future Datetime", from_tz=None):
+    time = datetime(entry, thing_name)
+    if time < _dt.datetime.utcnow():
+        raise ValueError(f"That {thing_name} is in the past! Must give a Future datetime!")
+    return time
+
+
+def signed_integer(entry, thing_name="Signed Integer"):
+    if not entry:
+        raise ValueError(f"Must enter a whole number for {thing_name}!")
+    try:
+        num = int(entry)
+    except ValueError:
+        raise ValueError(f"Could not convert '{entry}' to a whole number for {thing_name}!")
+    return num
+
+
+def positive_integer(entry, thing_name="Positive Integer"):
+    num = signed_integer(entry, thing_name)
+    if not num >= 1:
+        raise ValueError(f"Must enter a whole number greater than 0 for {thing_name}!")
+    return num
+
+
+def unsigned_integer(entry, thing_name="Unsigned Integer"):
+    num = signed_integer(entry, thing_name)
+    if not num >= 0:
+        raise ValueError(f"{thing_name} must be a whole number greater than or equal to 0!")
+    return num
+
+
+def boolean(entry, thing_name="True/False"):
+    entry = entry.upper()
+    error = f"Must enter 0 (false) or 1 (true) for {thing_name}. Also accepts True, False, On, Off, Yes, No, Enabled, and Disabled"
+    if not entry:
+        raise ValueError(error)
+    if entry in ('1', 'TRUE', 'ON', 'ENABLED', 'ENABLE', 'YES'):
+        return True
+    if entry in ('0', 'FALSE', 'OFF', 'DISABLED', 'DISABLE', 'NO'):
+        return False
+    raise ValueError(error)
+
+
+def timezone(entry, thing_name="Timezone"):
+    if not entry:
+        raise ValueError(f"No {thing_name} entered!")
+    found = _partial(_TZ_DICT.keys(), entry)
+    if found:
+        return _TZ_DICT[found]
+    raise ValueError(f"Could not find timezone '{entry}' for {thing_name}!")
+
+
+def email(entry, thing_name="Email Address"):
+    try:
+        _val_email(entry) #offloading the hard work to Django!
+    except _error:
+        raise ValueError(f"That isn't a valid {thing_name}!")
+    return entry
+
+
+def lock(entry, thing_name='lockstring', options=None):
+    entry = entry.strip()
```

Comment:
Btw, this will fail if entry is not of the right form (for example it'll be an `AttributeError` if entry is a number). For these to be proper validators you Make sure the validator fails deliberately with a ValuError when input is on the wrong form.